pytest_picked/modes.py

```diff
@@ -26,6 +25,45 @@ def affected_tests(self):
                     files.append(file_or_folder)
         return files, folders
 
+    def only_tests(self):
+        raw_output = self.git_output()
+        re_list = [
+            item.replace(".", r"\.").replace("*", ".*")
+            for item in self.test_file_convention
+        ]
+        test_file_regex = r"(\/|^)" + r"|".join(re_list)
+        test_class_regex = r"(?<=class\s).*Test.*.[\w](?=\(|\:)"
+        test_regex = r"(?<=def\s)test.*(?=\(|/:)"
+
+        tests = []
+        last_file_name = None
+        file_dict = {}
+        class_name = ""
+        for candidate in raw_output.splitlines():
+            file_or_test = self.parser(candidate)
+            if file_or_test:
+                if re.search(test_file_regex, file_or_test):
+                    last_file_name = file_or_test
+                    if last_file_name not in file_dict.keys():
+                        file_dict[last_file_name] = []
+                if re.search(test_class_regex, file_or_test):
```

Comment:
The way is implemented you'd have the test name and the class among the tests that will be executed. Will it be executed twice? What are your thoughts on it? To reproduce this situation you can create a new test in a class in a module that already exists.

Reply:
I have now modified the logic such that we will directly be running all the tests from the changed test class, segregating only changed tests from test class not possible in call cases as git diff falls part on the test class level. I even updated the tests with duplicate class and multiple tests from changed test class does not affect the logic